controllers/pkg/reconcilers/bootstrap-packages/fake_client.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 The Nephio Authors
```

Comment:
this file should move the porch client library iso in the bootstrap pkg

Reply:
I did consider this but thought this fake is quite specific to this controller. If another controller requires a mock client, their impl would differ so should this not live next to the controller it's mocking the client for?

Reply:
I believe it is better to make it generic, so other people can consume it. I did the same for ipam and clan an tried to build in generic capabilities to satisfy multiple consumers. We otherwise have to maintain all this code and it is all over the place.

Reply:
I will need some assistance here. The fake is for any controller runtime client really so not porch specific. I have hacked a solution to load the response data from files but needs to be refined/refactored.